modules/google-analytics/classes/wp-google-analytics-universal.php

```diff
@@ -195,23 +199,187 @@ public function loop_add_to_cart() {
+					'id': productSku ? productSku : '#' + productID,
+					'quantity': $( this ).data( 'quantity' ),
+				};
+				ga( 'ec:addProduct', productDetails );
+				ga( 'ec:setAction', 'add' );
+				ga( 'send', 'event', 'UX', 'click', 'add to cart' );
 			} );"
 		);
 	}
 
-	public function wp_footer() {
+	public function remove_from_cart() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_remove_from_cart_is_enabled() ) {
+			return;
+		}
+
+		wc_enqueue_js(
+			"$( '.remove' ).click( function() {
+				var productSku = $( this ).data( 'product_sku' );
+				var productID = $( this ).data( 'product_id' );
+				var quantity = $( this ).parent().parent().find( '.qty' ).val()
+				var productDetails = {
+					'id': productSku ? productSku : '#' + productID,
+					'quantity': quantity ? quantity : '1',
+				};
+				ga( 'ec:addProduct', productDetails );
+				ga( 'ec:setAction', 'remove' );
+				ga( 'send', 'event', 'UX', 'click', 'remove from cart' );
+			} );"
+		);
+	}
+
+	/**
+	* Adds the product ID and SKU to the remove product link (for use by remove_from_cart above) if not present
+	*/
+	public function remove_from_cart_attributes( $url, $key ) {
+		if ( false !== strpos( $url, 'data-product_id' ) ) {
+			return $url;
+		}
+
+		$item = WC()->cart->get_cart_item( $key );
+		$product = $item[ 'data' ];
+
+		$new_attributes = sprintf( 'href="%s" data-product_id="%s" data-product_sku="%s"',
+			esc_attr( $url ),
+			esc_attr( $product->get_id() ),
+			esc_attr( $product->get_sku() )
+			);
+		$url = str_replace( 'href=', $new_attributes );
+		return $url;
+	}
+
+	public function listing_impression() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_product_impressions_is_enabled() ) {
+			return;
+		}
+
+		global $product, $woocommerce_loop;
+		if ( isset( $_GET['s'] ) ) {
+			$list = "Search Results";
+		} else {
+			$list = "Product List";
+		}
+
+		$item_details = array(
+			'id' => $product->get_id(),
+			'name' => $product->get_title(),
+			'category' => Jetpack_Google_Analytics_Utils::get_product_categories_concatenated( $product ),
+			'list' => $list,
+			'position' => $woocommerce_loop['loop']
+		);
+		wc_enqueue_js( "ga( 'ec:addImpression', " . wp_json_encode( $item_details ) . " );" );
+	}
+
+	public function listing_click() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_product_clicks_is_enabled() ) {
+			return;
+		}
+
+		if ( isset( $_GET['s'] ) ) {
+			$list = "Search Results";
+		} else {
+			$list = "Product List";
+		}
+
+		global $product, $woocommerce_loop;
+
+		$selector = ".products .post-" . esc_js( $product->get_id() ) . " a";
+
+		$item_details = array(
+			'id' => $product->get_id(),
```

Comment:
Should this be `sku or id` like the other entries?

Reply:
Fixed in https://github.com/Automattic/jetpack/pull/8218/commits/81f4fd874ba68294405482ddbaefac9d8477f541